pkg/controller/apicast/apicast_logic_reconciler.go

```diff
@@ -286,6 +286,38 @@ func (r *APIcastLogicReconciler) UserProvidedSecretResourceVersionAnnotations(us
 	return annotations
 }
 
+func (r *APIcastLogicReconciler) APIcastFromCRContents() (*apicast.APIcast, error) {
```

Comment:
This method has been implemented in order to be able to obtain an APIcast instance from outside the reconciler code.
Notice how we get the user provided secrets (and just get, not reconcile them like we do in the Reconcile method)

Reply:
this sounds like a function comment, not a PR comment